platform/security/handler/security-handler-oidc/src/main/java/org/codice/ddf/security/oidc/client/HandlerConfiguration.java

```diff
@@ -26,85 +27,55 @@
 import org.pac4j.oidc.profile.creator.OidcProfileCreator;
 import org.pac4j.oidc.profile.google.GoogleOidcProfile;
 import org.pac4j.oidc.profile.keycloak.KeycloakOidcProfile;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class HandlerConfiguration {
-
-  private OidcClient oidcClient;
-
-  private OidcConfiguration oidcConfiguration;
-
   private static final String IDP_TYPE = "idpType";
-
   private static final String CLIENT_ID = "clientId";
-
+  private static final String REALM = "realm";
   private static final String SECRET = "secret";
-
-  private static final String DISCOVERY_URI = "discoveryURI";
-
+  private static final String DISCOVERY_URI = "discoveryUri";
+  private static final String BASE_URI = "baseUri";
   private static final String SCOPE = "scope";
-
   private static final String USE_NONCE = "useNonce";
-
   private static final String RESPONSE_TYPE = "responseType";
-
   private static final String RESPONSE_MODE = "responseMode";
+  private static final String LOGOUT_URI = "logoutUri";
 
-  private static final String REALM = "realm";
+  private static final Logger LOGGER = LoggerFactory.getLogger(HandlerConfiguration.class);
 
-  private static final String BASE_URI = "baseUri";
+  private OidcClient oidcClient;
+  private OidcConfiguration oidcConfiguration;
 
-  private static final String LOGOUT_URI = "logoutUri";
+  private String idpType;
+  private String clientId;
+  private String realm;
+  private String secret;
+  private String discoveryUri;
+  private String baseUri;
+  private String scope;
+  private boolean useNonce;
+  private String responseType;
+  private String responseMode;
+  private String logoutUri;
 
   private Map<String, Object> properties;
 
   private OidcLogoutActionBuilder logoutActionBuilder;
 
   private OidcProfileCreator oidcProfileCreator;
 
+  public void init(Map<String, Object> properties) {
+    this.properties = properties;
+    init();
+  }
+
   public void init() {
     if (properties == null) {
       return;
     }
-    if (properties.get(IDP_TYPE).equals("Keycloak")) {
```

Comment:
📘 This logic was refactored into the new `update` and `generateOidcClient` methods.